ansible/grouper/group_vars/all.yml

```diff
@@ -0,0 +1,3 @@
+---
+ns: qa
+grouper_gsh_image: harbor.cyverse.org/de/grouper:v3.0.0
```

Comment:
I interpreted the [variable precedence table](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#understanding-variable-precedence) to mean that playbook group variables have a higher precedence than inventory group variables. This value can still be overridden by specifying inventory group variables for a specific group, but I was afraid that it might cause confusion. Would it be reasonable to move some of these tasks to a role and have the image defined in variable defaults instead?

Also, would it be reasonable to define it like this?

```
grouper_gsh_image: "harbor.cyverse.org/de/grouper:{{grouper_service_version}}"
```

If something like that would work then it would be possible to specify just the service version for sites that are running a different image version.

Reply:
Yeah, I went back and forth on this and couldn't decide. I could do that, or I could do what was done in other playbooks and use `_var_name` in the playbook and allow it to be overridden by a `var_name` variable by setting the value to `"{{ var_name | default('sane default') }}"`. I noticed that the postgresql roles/playbooks do that, which  isn't a bad solution.

I should still be able to split out the version number, either way.

Reply:
I've moved logic for the grouper init into a role and set the variable in the defaults. I also split out the version into a separate var and had change the container image definition to use it in the play.